compiler/src/main/antlr/Cellmata.g4

```diff
@@ -52,10 +52,12 @@ decrement_stmt
 return_stmt : STMT_RETURN expr END ;
 
 //For-loop
-for_stmt : STMT_FOR PAREN_START for_variable? END for_condition END for_increment? PAREN_END code_block ;
-for_variable : assignment ;
+for_stmt : STMT_FOR PAREN_START for_init? END for_condition END for_post_iteration? PAREN_END code_block ;
+for_init : assignment ;
 for_condition : expr ;
-for_increment : expr | assignment ;
+for_post_iteration : assignment ;
```

Comment:
if expr is omitted here, increment (i.e i-- or i++) is not possible in for loop as an update

Reply:
That is correct. We are removing unary increment and decrement soon.